packages/web-app-text-editor/src/index.js

```diff
@@ -19,41 +21,69 @@ const routes = [
   }
 ]
 
-const fileExtensionConfig = {
-  canBeDefault: true
+const fileExtensions = () => {
+  const extensions = [
+    {
+      extension: 'txt',
+      label: $gettext('Plain text file')
+    },
+    {
+      extension: 'md',
+      label: $gettext('Markdown file')
+    },
+    {
+      extension: 'js',
+      label: $gettext('JavaScript file')
+    },
+    {
+      extension: 'json',
+      label: $gettext('JSON file')
+    },
+    {
+      extension: 'xml',
+      label: $gettext('XML file')
+    },
+    {
+      extension: 'py',
+      label: $gettext('Python file')
+    },
+    {
+      extension: 'php',
+      label: $gettext('PHP file')
+    },
+    {
+      extension: 'yaml',
+      label: $gettext('YAML file')
+    }
+  ]
+  let blackList =
+    window.Vue.$store.getters.extensionConfigByAppId(appId).blockedFileExtensions || []
```

Comment:
This seems a workaround. I hope in the future we can have the callables I mention in the other ticket.

Reply:
Yes, that's clearly a workaround. We want to provide the feature for you guys, but we can't offer it in a technically nice way right now, as the extension framework is not on the short term roadmap.